src/main/java/blusunrize/immersiveengineering/common/world/IEWorldGen.java

```diff
@@ -50,15 +53,44 @@ public static void addOreGen(String name, BlockState state, int maxVeinSize, int
 		for(Biome biome : Biome.BIOMES)
 			biome.addFeature(Decoration.UNDERGROUND_ORES, feature);
 		features.put(name, feature);
+
+		ConfiguredFeature<?> retroFeature = Biome.createDecoratedFeature(IEContent.ORE_RETROGEN, cfg, Placement.COUNT_RANGE,
+				new CountRangeConfig(chunkOccurence, minY, minY, maxY));
+		retroFeatures.put(name, retroFeature);
 	}
 
 	public void generateOres(Random random, int chunkX, int chunkZ, World world, boolean newGeneration)
 	{
 		if(!oreDimBlacklist.contains(world.getDimension().getType().getRegistryName()))
-		for(Entry<String, ConfiguredFeature<?>> gen : features.entrySet())
-			if(newGeneration||retrogenMap.get("retrogen_"+gen.getKey()))
-				gen.getValue().place(world, world.getChunkProvider().getChunkGenerator(),
-						random, new BlockPos(16*chunkX, 0, 16*chunkZ));
+		{
+			if(newGeneration)
+			{
+				for(Entry<String, ConfiguredFeature<?>> gen : features.entrySet())
+				{
+
+					gen.getValue().place(world, world.getChunkProvider().getChunkGenerator(), random, new BlockPos(16*chunkX, 0, 16*chunkZ));
+				}
+			}
+			else
+			{
+				for(Entry<String, ConfiguredFeature<?>> gen : retroFeatures.entrySet())
+				{
+					boolean retrogen = false;
+					try
+					{
+						retrogen = retrogenMap.get("retrogen_"+gen.getKey());
+					} catch(NullPointerException e)
+					{
+						retrogen = false;
+					}
```

Comment:
This block seems strange... Why are you catching NPEs rather than just checking whether something is `null`?

Reply:
no reason, just leftover from debugging, as `retrogenMap` was completely empty before this pr